test/junit/scala/collection/generic/DecoratorsTest.scala

```diff
@@ -0,0 +1,243 @@
+    implicit def SeqDecorator[Repr](coll: Repr)(implicit seq: IsSeqLike[Repr]): SeqDecorator[Repr, seq.type] =
+      new SeqDecorator(coll)(seq)
+    class SeqDecorator[Repr, S <: IsSeqLike[Repr]](coll: Repr)(implicit val seq: S) {
+      def groupedWith[Group, That](p: seq.A => Boolean)(implicit
+        group: BuildFrom[Repr, seq.A, Group],
+        bf: BuildFrom[Repr, Group, That]
+      ): That = {
+        val `this` = seq(coll)
+
+        val groups = bf.newBuilder(coll)
+        val it = `this`.iterator
+
+        var currentGroup = group.newBuilder(coll)
+        var lastTestResult = Option.empty[Boolean]
+
+        while (it.hasNext) {
+          val elem = it.next()
+          val currentTest = p(elem)
+
+          lastTestResult match {
+            case None =>
+              currentGroup.addOne(elem)
+            case Some(lastTest) if currentTest == lastTest =>
+              currentGroup.addOne(elem)
+            case Some(_) =>
+              groups.addOne(currentGroup.result())
+              currentGroup = group.newBuilder(coll).addOne(elem)
+          }
+
+          lastTestResult = Some(currentTest)
+        }
+
+        groups.addOne(currentGroup.result()).result()
+      }
+    }
+
+    val p: Int => Boolean = _ % 2 == 0
+    val xs = List(1, 2, 3).groupedWith(p)
+    val xsT: List[List[Int]] = xs
+    val xs2 = List(1, 2, 3).view.groupedWith(p)
+    val xs2T: View[View[Int]] = xs2
+    val xs3 = Vector(1, 2, 3).groupedWith(p)
+    val xs3T: Vector[Vector[Int]] = xs3
+    val xs4 = Vector(1, 2, 3).view.groupedWith(p)
+    val xs4T: View[View[Int]] = xs4
+    val xs5 = (1 to 10).groupedWith(p)
+    val xs5T: IndexedSeq[IndexedSeq[Int]] = xs5
+    val xs6 = (1 to 10).view.groupedWith(p)
+    val xs6T: View[View[Int]] = xs6
+    val xs7 = Array(1, 2, 3).groupedWith(p)
+    val xs7T: Array[Array[Int]] = xs7
+    val xs8 = Array(1, 2, 3).view.groupedWith(p)
+    val xs8T: View[View[Int]] = xs8
+    val xs9 = "foo".groupedWith(_.isLower)
+    val xs9T: IndexedSeq[String] = xs9
+    val xs10 = "foo".view.groupedWith(_.isLower)
+    val xs10T: View[View[Char]] = xs10
+  }
+
+  @Test
+  def mapDecorator: Unit = {
+
+    implicit def MapDecorator[Repr](coll: Repr)(implicit map: IsMapLike[Repr]): MapDecorator[Repr, map.type] =
+      new MapDecorator(coll)(map)
+
+    class MapDecorator[C, M <: IsMapLike[C]](coll: C)(implicit val map: M) {
+      def leftOuterJoin[W, That](other: Map[map.K, W])(implicit bf: BuildFrom[C, (map.K, (map.V, Option[W])), That]): That = {
+        val b = bf.newBuilder(coll)
+        for ((k, v) <- map(coll)) {
+          b += k -> (v, other.get(k))
+        }
+        b.result()
+      }
+      def rightOuterJoin[W, That](other: Map[map.K, W])(implicit bf: BuildFrom[C, (map.K, (Option[map.V], W)), That]): That = {
+        val b = bf.newBuilder(coll)
+        for ((k, w) <- other) {
+          b += k -> (map(coll).get(k), w)
+        }
+        b.result()
+      }
+    }
```

Comment:
I’ve tried to write a decorator that uses `MapOps` instead of `IsMapLike` but it doesn’t work:

~~~ scala
    implicit class MapDecorator[M <: MapOps[K, V, CC, C], K, V, CC[X, Y] <: IterableOps[_, AnyConstr, _], C](coll: M) {
      def leftOuterJoin[W, That](other: Map[K, W])(implicit bf: BuildFrom[M, (K, (V, Option[W])), That]): That = {
        val b = bf.newBuilder(coll)
        for ((k, v) <- coll) {
          b += k -> (v, other.get(k))
        }
        b.result()
      }
      def rightOuterJoin[W, That](other: Map[K, W])(implicit bf: BuildFrom[M, (K, (Option[V], W)), That]): That = {
        val b = bf.newBuilder(coll)
        for ((k, w) <- other) {
          b += k -> (coll.get(k), w)
        }
        b.result()
      }
    }
~~~

The implicit conversion is never triggered. If I try to explicitly call the conversion, I get type errors like so:

~~~ scala
[error] inferred type arguments [scala.collection.immutable.Map[Int,String],Nothing,Nothing,Nothing,Nothing] do not conform to method MapDecorator's type parameter bounds [M <: scala.collection.MapOps[K,V,CC,C],K,V,CC[X, Y] <: scala.collection.IterableOps[_, collection.AnyConstr, _],C]
[error]     val mapLJoin = MapDecorator(map).leftOuterJoin(Map(1 -> "bar"))
[error]                    ^
~~~

I think users will need to define an intermediate type for the type unification to work. This is exactly what `IsMapLike` does, so why not just keep it instead of telling users to copy it everywhere?